### PR TITLE
Add: Time Conversion Function

### PR DESCRIPTION
**Pull request by JeelGajera**
### Describe your change:
This PR add the Time unit conversion function. 
conversion_chart
[chart source](https://en.wikipedia.org/wiki/Unit_of_time)

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by JeelGajera** [comment]
@cclauss please review this PR.

File: conversions/time_conversions.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+"""
+A unit of time is any particular time interval, used as a standard
+way of measuring or expressing duration.
+The base unit of time in the International System of Units (SI),
+and by extension most of the Western world, is the second,
+defined as about 9 billion oscillations of the caesium atom.
+
+WIKI: https://en.wikipedia.org/wiki/Unit_of_time
```
Comment (cclauss) [inline conversions/time_conversions.py:8]: Let's use our 88 characters per line...
```suggestion
A unit of time is any particular time interval, used as a standard way of measuring or
expressing duration.  The base unit of time in the International System of Units (SI),
and by extension most of the Western world, is the second, defined as about 9 billion
oscillations of the caesium atom.

https://en.wikipedia.org/wiki/Unit_of_time
```

File: conversions/time_conversions.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "weeks": 604800.0,  # 1 week=7d=168hr=10080min = 604800 sec
+}
+
+time_chart_inverse: dict[str, float] = {
+    "seconds": 1.0,
+    "minutes": 1 / 60.0,  # 1 minute = 1/60 hours
+    "hours": 1 / 3600.0,  # 1 hour = 1/3600 days
+    "days": 1 / 86400.0,  # 1 day = 1/86400 weeks
+    "weeks": 1 / 604800.0,  # 1 week = 1/604800 seconds
+}
```
Comment (cclauss) [inline conversions/time_conversions.py:25]: Let's calculate `time_chart_inverse`
```suggestion
time_chart_inverse: dict[str, float] = {key: 1 / value for key, value in time_chart.items()}
```

File: conversions/time_conversions.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    >>> convert_time(3600, "seconds", "hours")
+    1.0
+    >>> convert_time(1, "days", "hours")
+    24.0
+    >>> convert_time(120, "minutes", "seconds")
+    7200.0
+    >>> convert_time(2, "weeks", "days")
+    14.0
+    >>> convert_time(0.5, "hours", "minutes")
+    30.0
```
Comment (cclauss) [inline conversions/time_conversions.py:41]: Test should show how the algorithm fails...
```suggestion
    30.0
    >>> convert_time(-3600, "seconds", "hours")
    >>> convert_time("Hello", "hours", "minutes")
    >>> convert_time([0, 1, 2], "weeks", "days")
    >>> convert_time(50, "HOURS", "days")
    >>> convert_time(50, "hours", "YEARS")
```

File: conversions/time_conversions.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "weeks": 1 / 604800.0,  # 1 week = 1/604800 seconds
+}
+
+
+def convert_time(time_value: float, unit_from: str, unit_to: str) -> float:
+    """
+    Convert time from one unit to another using the time_chart above.
+
+    >>> convert_time(3600, "seconds", "hours")
+    1.0
```
Comment (cclauss) [inline conversions/time_conversions.py:33]: Let's show that some results will not be whole numbers...
```suggestion
    1.0
    >>> convert_time(3500, "seconds", "hours")
    ???
```

File: conversions/time_conversions.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        raise ValueError(msg)
+    return round(time_value * time_chart[unit_from] * time_chart_inverse[unit_to], 3)
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+
+    print(convert_time(3600, "seconds", "hours"))
```
Comment (cclauss) [inline conversions/time_conversions.py:57]:
```suggestion
    print(f"{convert_time(3600, "seconds", "hours") = }")
```
Please add one or two more of these.

**Review by cclauss — CHANGES_REQUESTED** [review]
Slick!!  A few more changes...

**Review by JeelGajera — COMMENTED** [review]
resolve the required changes

**Review by cclauss — APPROVED** [review]
A great way to pass the time!

**Comment by JeelGajera** [comment]
Please add hacktoberfest-accepted label so that it can be accepted in hacktoberfest.

**Comment by tianyizheng02** [comment]
@JeelGajera This repo has the `hacktoberfest` label, so there is no need to label individual PRs. Your PR will be accepted for Hacktoberfest even without the `hacktoberfest-accepted` label.

**Pull request merged**